"libzt" breaks the Haiku build process.

### Operating System

Other (please specify)

### DevilutionX version

1.4.1 (latest release)

### Describe

In file included from /boot/system/develop/headers/posix/net/if.h:9,
                 from /boot/system/develop/headers/posix/netinet/in.h:9,
                 from /boot/home/Desktop/devilutionX-1.4.1/build/_deps/libzt-src/ext/ZeroTierOne/ext/libnatpmp/getgateway.c:33:
/boot/home/Desktop/devilutionX-1.4.1/build/_deps/libzt-src/include/net/route.h:66:10: fatal error: sys/appleapiopts.h: No such file or directory
   66 | #include <sys/appleapiopts.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.

### To Reproduce

Just try to compile devilutionX 1.4.1 from the source tar.gz on Haiku x64.

### Expected Behavior

_No response_

### Additional context

_No response_

--- a/3rdParty/libzt/CMakeLists.txt
+++ b/3rdParty/libzt/CMakeLists.txt
@@ -5,7 +5,7 @@ set(BUILD_HOST_SELFTEST OFF)
 include(FetchContent)
 FetchContent_Declare(libzt
   GIT_REPOSITORY https://github.com/diasurgical/libzt.git
-  GIT_TAG 37a2efb0b925df632299ef07dc78c0af5f6b4756)
+  GIT_TAG 47e38259e9c99ba2d8e7241515fa1b56cb538c8c)
 FetchContent_MakeAvailableExcludeFromAll(libzt)
 
 if(NOT ANDROID)

